config/v1/types_infrastructure.go

```diff
@@ -1370,6 +1373,89 @@ type NutanixPlatformSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	PrismElements []NutanixPrismElementEndpoint `json:"prismElements"`
+
+	// failureDomains configures failure domains information for the Nutanix platform.
+	// If this is omitted, failure domains will not be used to spread Machines.
```

Comment:
Is this information only used for Machines? Is it used by the CCM or CSI at all?

How does this compare to upstream Cluster API implementations of failure domains for nutanix?

Reply:
Currently we don't plan to use the failure domains info in the nutanix-machine-controller. Adding the Nutanix failure domain configuration to the Infrastructure's NutanixPlatfromSpec, the purpose is to facilitate users to configure worker MachineSet spreading across the configured failure domains.
Keeping the two copies (in the Infrastrure and CCMS CRs) of the Nutanix failure domain configuration, the benefits is to allow the control plane nodes and worker nodes to use different failure domains, if it is required.

Reply:
So can you clarify for me where this information is used? How will the MachineSet controller consume this information?

Reply:
The configured failure domains in the Infrastructure's NutanixPlatformSpec can be referenced in the CPMS, Machine's nutanix providerSpec, and other places (CCM, CSI, etc.) when needed.